velox/connectors/hive/HivePartitionFunction.cpp

```diff
@@ -45,13 +45,16 @@ int32_t hashTimestamp(const Timestamp& ts) {
   return hashInt64((ts.getSeconds() << 30) | ts.getNanos());
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wreturn-type"
 template <TypeKind kind>
 inline uint32_t hashOne(
     const typename TypeTraits<kind>::NativeType& /* value */) {
```

Comment:
same comment. Could we use "[[noreturn]]" instead?

Reply:
This does not work here because hashOne is specialized for TypeKind::BOOLEAN, TypeKind::TINYINT, etc, and these specializations do return a value.

Reply:
A dummy return0 should work. Thanks.